extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts

```diff
@@ -613,12 +617,33 @@ export class VariantAnalysisManager
             });
           }
         };
-        await this.variantAnalysisResultsManager.download(
-          variantAnalysis.id,
-          repoTask,
-          this.getVariantAnalysisStorageLocation(variantAnalysis.id),
-          updateRepoStateCallback,
-        );
+        let retry = 0;
+        for (;;) {
+          try {
+            await this.variantAnalysisResultsManager.download(
+              variantAnalysis.id,
+              repoTask,
+              this.getVariantAnalysisStorageLocation(variantAnalysis.id),
+              updateRepoStateCallback,
+            );
+            break;
+          } catch (e) {
+            if (
+              retry++ < maxRetryCount &&
+              e instanceof FetchError &&
+              (e.code === "ETIMEDOUT" || e.code === "ECONNRESET")
+            ) {
+              void extLogger.log(
+                `Timeout while trying to download variant analysis with id: ${variantAnalysis.id}. Retrying...`,
```

Comment:
Can we also log the error message here? You can use the `getErrorMessage` function to convert the error into an error message to put in this log.